<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>678</width>
    <height>585</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>20</y>
      <width>101</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>process Number</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_1">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>50</y>
      <width>54</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Hole Size</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>20</y>
      <width>101</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Starting Address</string>
    </property>
   </widget>
   <widget class="QPushButton" name="uiTakePNum">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>60</y>
      <width>71</width>
      <height>27</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::StrongFocus</enum>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>Enter Num</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <property name="autoDefault">
     <bool>true</bool>
    </property>
    <property name="default">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="uiTakeHole">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>50</y>
      <width>85</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Enter Hole</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>130</y>
      <width>91</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>process Size</string>
    </property>
   </widget>
   <widget class="QPushButton" name="uiTakeProcess">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>130</y>
      <width>101</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Enter Process</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_1">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>50</y>
      <width>113</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_2">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>20</y>
      <width>113</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_3">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>20</y>
      <width>91</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_4">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>130</y>
      <width>113</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QTableView" name="tableView_1">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>170</y>
      <width>191</width>
      <height>291</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
   </widget>
   <widget class="QTableView" name="tableView_2">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>170</y>
      <width>241</width>
      <height>291</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>180</y>
      <width>81</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>Methodology</string>
    </property>
    <widget class="QPushButton" name="pushButton">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>71</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>First Fit</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>60</y>
       <width>71</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Best Fit</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_4">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>100</y>
       <width>71</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>Worst Fit</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>90</y>
      <width>361</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>NOTE press &quot;EnterKey&quot; only when you write BOTH variables</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>500</y>
      <width>85</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset All</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_5">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>500</y>
      <width>91</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Compaction</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_6">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>480</y>
      <width>85</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Merge</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_7">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>130</y>
      <width>141</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>De-Allocate Process</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>320</y>
      <width>121</width>
      <height>17</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Choose ONE ONLY </string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>678</width>
     <height>27</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>lineEdit_3</tabstop>
  <tabstop>uiTakePNum</tabstop>
  <tabstop>lineEdit_2</tabstop>
  <tabstop>lineEdit_1</tabstop>
  <tabstop>uiTakeHole</tabstop>
  <tabstop>lineEdit_4</tabstop>
  <tabstop>uiTakeProcess</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>pushButton_2</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>uiTakePNum</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>enterNum()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>562</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>586</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiTakeHole</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>enterHole()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>356</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>383</x>
     <y>108</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiTakeProcess</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>enterProcess()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>347</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>314</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_3</sender>
   <signal>returnPressed()</signal>
   <receiver>uiTakePNum</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>549</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>558</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_4</sender>
   <signal>returnPressed()</signal>
   <receiver>uiTakeProcess</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>210</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>297</x>
     <y>175</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>firstFit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>310</x>
     <y>266</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>bestFit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>304</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>274</x>
     <y>293</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_4</sender>
   <signal>returnPressed()</signal>
   <receiver>lineEdit_4</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>154</x>
     <y>191</y>
    </hint>
    <hint type="destinationlabel">
     <x>156</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiTakeProcess</sender>
   <signal>released()</signal>
   <receiver>lineEdit_4</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>281</x>
     <y>187</y>
    </hint>
    <hint type="destinationlabel">
     <x>258</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiTakeHole</sender>
   <signal>released()</signal>
   <receiver>lineEdit_1</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>291</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>uiTakeHole</sender>
   <signal>released()</signal>
   <receiver>lineEdit_2</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>281</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_2</sender>
   <signal>returnPressed()</signal>
   <receiver>uiTakeHole</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>198</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>313</x>
     <y>108</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_1</sender>
   <signal>returnPressed()</signal>
   <receiver>uiTakeHole</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>102</y>
    </hint>
    <hint type="destinationlabel">
     <x>326</x>
     <y>98</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>657</x>
     <y>515</y>
    </hint>
    <hint type="destinationlabel">
     <x>194</x>
     <y>462</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>merge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>464</x>
     <y>500</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>507</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_7</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>deAllocate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>528</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>564</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>handleButton()</slot>
  <slot>takeNum()</slot>
  <slot>enterNum()</slot>
  <slot>enterHole()</slot>
  <slot>enterProcess()</slot>
  <slot>firstFit()</slot>
  <slot>bestFit()</slot>
  <slot>reset()</slot>
  <slot>merge()</slot>
  <slot>deAllocate()</slot>
 </slots>
</ui>
